### ISSUE TITLE
Cancelling the InventoryTransactionEvent (on SlotChange) doesn't work correctly.

### ISSUE THREAD

**Issue by PocketMiner82** [GameModsBR/PowerNukkit#343]
# 🐞 I found a bug
I'm using FakeInventories, cancelling the event worked before 1.16. ~~Maybe FakeInventories issue, idk.~~ Nope, no FakeInventory issue. Also happens when cancelling "normal" InventoryTransactionEvent, e.g. EnderChest SlotChange.

### 📸 Screenshots / Videos
https://advancedcraft.net/vid/Bug_SlotChange.mp4

### ▶ Steps to Reproduce
Create a FakeInventory, cancel the FakeSlotChangeEvent.
Edit: Or just cancel the InventoryTransactionEvent.

### ✔ Expected Behavior
The item should also get deleted from the mouse.

### ❌ Actual Behavior
The Item is still in at the mouse.

**Comment by MrRazamataz** [comment]
What plugin are you using to interact with the inventory?

**Comment by PocketMiner82** [comment]
Own plugin: Code to cancel interaction:
```java
if (ev.getAction().getSourceItem().hasCompoundTag())
    if (ev.getAction().getSourceItem().getNamedTag().exist("canPutOut") && !ev.getAction().getSourceItem().getNamedTag().getBoolean("canPutOut")) {
        ev.setCancelled();
        return;
    }
```

Following event (FakeInventories): ` public static void onFakeInventoryChange(FakeSlotChangeEvent ev)`

**Comment by PocketMiner82** [comment]
The item has the tag set to false, it worked before the update to 1.16.

**Comment by MrRazamataz** [comment]
Is this a private plugin or is it one that I could download because currently I have a plugin that only allows me to view them and not control them and I would like to be able to control them too...

**Comment by PocketMiner82** [comment]
Private, sorry

**Comment by PocketMiner82** [comment]
Fixed with PR: #349.

**Comment by joserobjr** [comment]
Final fix at #384
